lib/src/coap_client.dart

```diff
@@ -424,43 +423,39 @@ class CoapClient {
     await _lock.synchronized(() async {
       // Set endpoint if missing
       if (_endpoint == null) {
-        final destination =
-            await _lookupHost(uri.host, addressType, bindAddress);
+        final destination = await _lookupHost(uri.host, addressType);
         final socket = CoapINetwork.fromUri(
           uri,
           address: destination,
+          bindAddress: bindAddress,
+          addressType: addressType,
```

Comment:
Great changes! (Especially regarding TCP support \o/)

Just a quick comment: I think the `addressType` could probably be derived from the `destination` address?

By the way, should we change the default `addressType` for lookups to `InternetAddressType.any`? Or could this cause problems for environments where IPv6 is unavailable?

Reply:
Good points, fixed in my latest commit! I think `InternetAddressType.any` should be fine? Worst-case it can be overridden by the user. Changing didn't cause any problems for me at least.

Still need to test things out a bit more, especially the retry logic for TCP. IIRC futures (including timers) are permanently killed when an app goes to sleep in Flutter for example, at least on iOS. On the other hand, perhaps then the entire `CoapClient` should be re-created? In my Flutter app I'm listening for `AppLifecycleState.resumed` to get things going again when the app resumes from sleep. It might not be something for this library to worry about I mean (other than perhaps mentioning it in the README).

I'm also firing a `CoapDataReceivedEvent` even if `d.data.isEmpty` now, mainly for debugging (if we receive empty datagrams from the server it could be good to know). Need to test that it doesn't break anything.

Reply:
Fixed some potential issues, updated the README to include a note on futures in Flutter (and to make it a bit less dense). Squashed the commits.

If you can, please try the DTLS code to see that it works okay (best of all by cutting the connection during requests, for example using the code snippet in https://github.com/shamblett/coap/issues/110, and then turning it back on again... can take a little while before it comes back up).